src/mono/mono/metadata/assembly.h

```diff
@@ -122,6 +122,14 @@ typedef struct {
 	unsigned int size;
 } MonoBundledAssembly;
 
+typedef struct WasmAssembly_ WasmAssembly;
+
+struct WasmAssembly_ {
```

Comment:
If this is wasm-specific, I don't think it belongs here.

Also a nit: our convention is to either have the struct name be the same or with a prefixed underscore, not a postfixed one.